### ISSUE TITLE
Support of AMD format?

### ISSUE THREAD

**Issue by wangjia184** [PepsRyuu/nollup#221]
```
Error: Invalid format "amd". Only es, cjs, iife supported.
    at validateOutputFormat (C:\Users\Jerry.Wang\AppData\Roaming\npm\node_modules\nollup\lib\impl\RollupConfigContainer.js:41:15)
    at RollupConfigContainer.setOutputOptions (C:\Users\Jerry.Wang\AppData\Roaming\npm\node_modules\nollup\lib\impl\RollupConfigContainer.js:131:13)
    at NollupContext.setOutputOptions (C:\Users\Jerry.Wang\AppData\Roaming\npm\node_modules\nollup\lib\impl\NollupContext.js:202:21)
    at Object.generate (C:\Users\Jerry.Wang\AppData\Roaming\npm\node_modules\nollup\lib\index.js:41:21)
    at generateBundles (C:\Users\Jerry.Wang\AppData\Roaming\npm\node_modules\nollup\lib\dev-middleware.js:88:47)
    at C:\Users\Jerry.Wang\AppData\Roaming\npm\node_modules\nollup\lib\dev-middleware.js:187:9
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
```

Can I use HMR with AMD modules?

**Comment by PepsRyuu** [comment]
I haven't coded any support for ```amd``` modules. 

Is there a particular reason the other formats aren't suitable for your use case? Can you describe the use case in more detail?

**Comment by wangjia184** [comment]
@PepsRyuu  My project  is not a tranditional web application.  It provides widgets to be loaded into other's websites dynamically.  This means AMD is more suitable in my case

**Comment by PepsRyuu** [comment]
Gotcha. It wouldn't be too difficult to add in AMD support as an output. My only concern would be the HMR side of things. I will need to assess the feasibility of this, but will need a bit of time to look into it. :)

**Comment by PepsRyuu** [comment]
First pass at supporting AMD: https://github.com/PepsRyuu/nollup/commit/d7efe143e365ccd07feeb7a6282bb8244a7e4adc

Would be great to get your thoughts on it and see if this solution works for you. Still need to do quite a bit of refactoring to support this properly along with several test cases.

**Comment by PepsRyuu** [comment]
Just a small update, this is going to require quite a bit of time to properly test, as I need to do a lot of refactoring to add this in. It will be a while still before a release.

**Comment by PepsRyuu** [comment]
Released in ```0.19.0```.
